pkg/logger/gorm_test.go

```diff
@@ -96,28 +129,55 @@ func TestNewGormLogger(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			var fields Fields
+			// var fields Fields
 			var buffer bytes.Buffer
 
 			b := NewBuilder(&tt.cfg)
 			l, err := b.BuildTestLogger(&buffer)
 			require.Nil(t, err)
 
-			gl := NewGormLogger(l)
-			gl.Print(tt.input...)
+			gormDB, mock := mockGormConnectionWithLogger(t, l)
 
-			if tt.isEmpty {
-				assert.Empty(t, buffer.Bytes())
+			if tt.resultError {
+				mock.ExpectExec(sampleQuery).WillReturnError(testDBError)
+			} else {
+				mock.ExpectExec(sampleQuery).WillReturnResult(
+					sqlmock.NewResult(expectedLastInsertID, expectedEffectedRows),
+				)
+			}
+
+			gormDB.Exec(sampleQuery)
+
+			if tt.isLogged {
+				assert.Regexp(t, tt.expected, buffer.String())
 			} else {
-				err = json.Unmarshal(buffer.Bytes(), &fields)
-				assert.Nil(t, err)
-
-				if tt.expected != nil {
-					assert.Equal(t, tt.expected, fields["sql"])
-				} else {
-					assert.NotEmpty(t, fields["message"])
-				}
+				assert.Empty(t, buffer.Bytes())
+			}
+
+			if err := mock.ExpectationsWereMet(); err != nil {
+				t.Errorf("there were unfulfilled expectations: %s", err)
 			}
 		})
+
+	}
+}
+
+func mockGormConnectionWithLogger(t *testing.T, l Logger) (*gorm.DB, sqlmock.Sqlmock) {
```

Comment:
[q]: is there a reason why we are introducing `sqlmock` here? Previously, we were defining unit tests for the `gormLogger`. Now it seems more like an integration testing.

Although, I don't see a big issue with mocking DBs, probably we can just keep do unit tests instead? Considering that [sqlmock is not in an active development](https://github.com/DATA-DOG/go-sqlmock/releases/tag/v1.5.0) atm (no releases for last 2 years)

Reply:
All functionality used here are within `logger` package so I don't really think it leans to be an integration test. Using `sqlmock` helps us cover more codes from top to button and actually mocks db behavior and it's logger together.